src/modules/design/diagram-detail/diagram-detail.ts

```diff
@@ -336,7 +337,29 @@ export class DiagramDetail {
       const xml: string = await this.bpmnio.getXML();
       this.activeDiagram.xml = xml;
 
-      await this.activeSolutionEntry.service.saveDiagram(this.activeDiagram);
+      const activeDiagramIsUnsavedDiagram: boolean = this.activeDiagram.uri.includes('temp-diagrams');
+
+      if (activeDiagramIsUnsavedDiagram && diagramPath) {
```

Comment:
Extract Business Rule

```suggestion
      const diagramPathSet: boolean = diagramPath !== undefined && diagramPath !== null;
      if (activeDiagramIsUnsavedDiagram && diagramPathSet) {
```